src/backend/distributed/planner/multi_planner.c

```diff
@@ -658,11 +659,29 @@ multi_join_restriction_hook(PlannerInfo *root,
 							JoinType jointype,
 							JoinPathExtraData *extra)
 {
+	JoinRestriction *joinRestriction = NULL;
 	JoinRestrictionContext *joinContext = NULL;
-	JoinRestriction *joinRestriction = palloc0(sizeof(JoinRestriction));
 	List *restrictInfoList = NIL;
+	MemoryContext restrictionsMemoryContext = CurrentRestrictionsMemoryContext();
+	MemoryContext oldMemoryContext = CurrentMemoryContext;
```

Comment:
It doesn't make sense to assign CurrentMemoryContext here, you're overwriting it below, so it's plainly misleading. I know we have that pattern in a bunch of places, but it's wrong.